docs: clarify Gateway.ExposeRoutingAPI (#10337)
ipfs/kubo: docs/config.md

### `Gateway.ExposeRoutingAPI`

An optional flag to expose Kubo `Routing` system on the gateway port

> as a [`/routing/v1`](https://specs.ipfs.tech/routing/routing-v1/) endpoint on `127.0.0.1`.

Reviewer: ```suggestion
as an [HTTP `/routing/v1`](https://specs.ipfs.tech/routing/routing-v1/) endpoint on `127.0.0.1`.
```

Use reverse proxy to expose it on a different hostname.

> This endpoint can be used by other Kubo instance, as illustrated in

Reviewer: ```suggestion
This endpoint can be used by other Kubo instances, as illustrated in
```

[`delegated_routing_v1_http_proxy_test.go`](https://github.com/ipfs/kubo/blob/master/test/cli/delegated_routing_v1_http_proxy_test.go).

> Kubo will skip routing results which are not actionable, for example, all

Reviewer: ```suggestion
Kubo will filter out routing results which are not actionable, for example, all
```
